### PR TITLE
ConnectivityCheck flags parallel way as connectable

### PR DESCRIPTION
**Pull request by vladlemberg**
### Description:

Please refer to https://github.com/osmlab/atlas-checks/issues/504

ConnectivityCheck scope is reduced by removing duplicated logic of crossing edges to EdgeCrossingEdge check.  

### Potential Impact:

Nodes that too close but connected to two+ edges won't be flagged. 

### Unit Test Approach:

Corresponded unit tests: invalidDisconnectedNodesCrossingTest, invalidDisconnectedEdgeCrossingTest  from ConnectivityCheck moved to EdgeCrossingEdge. 

### Test Results:
passed

### CODE REVIEW

File: src/test/java/org/openstreetmap/atlas/checks/validation/points/ConnectivityCheckTest.java [hunk 1]
```diff
@@ -55,13 +55,12 @@ public void invalidConnectedNodesTest()
...
     }
 
-    @Test
-    public void invalidDisconnectedEdgeCrossingTest()
-    {
-        this.verifier.actual(this.setup.invalidDisconnectedEdgeCrossingAtlas(),
-                new ConnectivityCheck(ConfigurationResolver.emptyConfiguration()));
-        this.verifier.globallyVerify(flags -> Assert.assertEquals(1, flags.size()));
-    }
+    /*
```
Comment (atiannicelli) [inline src/test/java/org/openstreetmap/atlas/checks/validation/points/ConnectivityCheckTest.java:58]: I think we should probably remove this (remove commented section) to simplify and clear up confusion. If we leave it in there people may be confused as to why they are there.

**Review by atiannicelli — DISMISSED** [review]
Looks good to me. I would remove the commented out sections, though. 

Also, just to be clear, there is one type of connectivity that we will no longer be detected if we make this change:
- When a node in a road comes close to another node in another road but the two ways don't intersect.

I think that this is GOOD and is specifically why we started making this change because we noticed that parallel bridges were being flagged incorrectly, but I just wanted to call it out.

**Comment by atiannicelli** [comment]
Also, @vladlemberg do you have any stats about how many issues the new version of the check finds compared to the original?
Looks like currently there are 4253 flags in USA. How many after the change? (You can pick any country - just looking for stats.)

**Comment by vladlemberg** [comment]
@atiannicelli, @Bentleysb, @sayas01, 
Number of cases in USA have been reduced from 4K+ to 44! with the new logic. I have created private MR challenge. Please let me know if you want to take a look. Not sure whether or not that massive drop is good. 

My plan is to create MR with the reversed logic to ensure that it only comprises:
1) parallel ways.
2) very closed ways that should not be connected.
3) crossings (covered by edgeCrossingEdge)  

Please let me know if you have other thoughts.

**Review by atiannicelli — APPROVED** [review]
Looks good to me! Nice to see this get a bit more focussed.

**Comment by Bentleysb** [comment]
The unit test move is great, but has any analysis been done to check that the dropped flags are picked up my EdgeCrossingEdgeCheck?

**Comment by atiannicelli** [comment]
@Bentleysb we were thinking that if the tests that we moved over were caught by the edgecrossingedgecheck then that would be good enough. It is difficult to compare the two checks since they might flag a different OSM ID for each issue. For instance if there is an issue between way X and Way Y then the old connectivity check may have flagged way X and the edgeCrossingEdgecheck may have flagged Way Y. Do you have any suggestions about how to determine if the ECE Check is covering the issues that used to be flagged by the old connectivity check?

**Comment by Bentleysb** [comment]
@atiannicelli, I would suggest using the [Log Diff Command](https://github.com/osmlab/atlas-checks/blob/dev/src/main/java/org/openstreetmap/atlas/checks/commands/AtlasChecksLogDiffSubCommand.java) to compare a run with and without the changes. Once you have the diffs from the runs compare the CC and ECE diffs geographically to see if they are ruffly in the same spots. Then spot check flags to see if the nodes flagged by CC have an Edge they're connected to flagged in ECE. This doesn't need to be be comprehensive, but it would be nice have a once over the diffs of real world data. There are just too many weird scenarios in OSM for unit tests to cover everything.

**Comment by Bentleysb** [comment]
Closing as a newer PR was merged.

**Pull request closed without merging**